org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -412,21 +469,61 @@ class CentralController:
 
     def __init__(self):
         self.concState = $!{rootState.getName()}_State(None, None)
-        self.commands = {"help": self.printHelp, "states": self.printStates, "execute": self.execute}
+        self.concState.default_initialize()
+        self.commands = {"help": self.printHelp, "execute": self.execute}
         #if(${allEnvEvents.size()}> 0)
         self.possibleEvents = {"${allEnvEvents.get(0).getModifiedName()}"#foreach($event in ${allEnvEvents.subList(1, ${allEncEvents.size()})}), "$!{event.getModifiedName()}"#end}
         #else
         self.possibleEvents = {}
         #end
-
+    
+    def handle_event(self):
+        unused_conc_states = self.concState.get_conc_descendants()
+        old_active_states = dict()
+        for conc_state in unused_conc_states:
+            old_active_states[conc_state.__repr__()] = conc_state.get_active_child_state_repr()
+        prev_num_unused_conc_states = -1
+        
+        while len(unused_conc_states) > 0 and not prev_num_unused_conc_states == len(unused_conc_states):
+            prev_num_unused_conc_states = len(unused_conc_states)
+            state_queue = [self.concState]
+            while state_queue:
+                cur_state = state_queue[0]
+                del state_queue[0]
+
+                possible_transitions = []
+                for k, v in cur_state.transitions.items():
+                    if k():
+                        possible_transitions.append(v)
+
+                if possible_transitions:
+                    chosen_transition = random.randint(0, len(possible_transitions) - 1)
+                    possible_transitions[chosen_transition]()
+                    # check if any concurrent states have taken a transition. If so,
+                    # remove concurrent states from the set to prevent additional transitions (big step maximality)
+                    for conc_state in unused_conc_states:
+                        if not conc_state.get_active_child_state_repr() == old_active_states[conc_state.__repr__()]: 
+                            unused_conc_states.remove(conc_state)
+                elif cur_state._is_children_conc:
+                    for conc_state in cur_state._children:
+                        if unused_conc_states.count(conc_state) > 0:
+                            state_queue.append(conc_state)
+                else:
+                    if cur_state.get_active_child_state() is not None:
```

Comment:
We can just combine this `if` statement with the `else` statement just above it. That is to merge lines 511 and 512 into `elif cur_state.get_active_child_state():`. Even though this is less rigorous than checking `is not None`, I think it provides better readability.